File: LeetCode 75 Study Plan to Ace Interviews/Q724 - Find Pivot Index/solution.cs
public class Solution {
    public int PivotIndex(int[] nums) {
        int sum = nums[0];
        for (int i = 1; i < nums.Length; i++) {
            sum += nums[i];
        }
        int leftSum = 0;
        for (int i = 0; i < nums.Length; i++) {
            int rightSum = sum - nums[i] - leftSum;
            if (leftSum == rightSum) {
                return i;
            }
            leftSum += nums[i];
        };
        return -1;
    }
}

File: LeetCode 75 Study Plan to Ace Interviews/Q724 - Find Pivot Index/solution.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        sum = 0
        for num in nums:
            sum += num
        leftSum = 0
        for index in range(len(nums)):
            rightSum = sum - leftSum - nums[index]
            if leftSum == rightSum:
                return index
            leftSum += nums[index]
        return -1
